### PR TITLE
Balance data with job manager

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/meta/processors/jobMan/StatisJobExecutor.h [hunk 1]
```diff
@@ -39,10 +39,10 @@ class StatisJobExecutor : public MetaJobExecutor {
...
 
 private:
-    // Statis job writes an additional data.
-    // The additional data is written when the statis job passes the check function.
-    // Update this additional data when job finishes.
-    kvstore::ResultCode save(const std::string& key, const std::string& val);
+    // // Statis job writes an additional data.
+    // // The additional data is written when the statis job passes the check function.
+    // // Update this additional data when job finishes.
+    // kvstore::ResultCode save(const std::string& key, const std::string& val);
```
Comment (critical27) [inline src/meta/processors/jobMan/StatisJobExecutor.h:45]: just delete it if you replace with `saveAdditionalData`

File: src/meta/processors/admin/LeaderBalanceProcessor.cpp [hunk 2]
```diff
@@ -6,17 +6,437 @@
...
+    auto ret = leaderBalance();
     handleErrorCode(ret);
     onFinished();
 }
 
+cpp2::ErrorCode LeaderBalanceProcessor::leaderBalance() {
+    folly::Promise<Status> promise;
+    auto future = promise.getFuture();
+    auto spaceInfoRet = MetaCommon::getSpaceInfo(kvstore_, space_);
+    if (!spaceInfoRet.ok()) {
```
Comment (critical27) [inline src/meta/processors/admin/LeaderBalanceProcessor.cpp:28]: We only support balance leader for one space? Emm, how to balance leader for all space?

**Review by critical27 — COMMENTED** [review]
Good job, several questions, would you post it in summary:
1. What is the new balance command? 
2. Does `balance leader` logic changed anything? Because I see you moved to another `LeaderBalanceProcessor.cpp`. If any changes, please tell me.

File: src/meta/processors/admin/LeaderBalanceProcessor.cpp (hunk 2 above)
Comment (darionyaphet) [reply src/meta/processors/admin/LeaderBalanceProcessor.cpp:28]: yep currently we only support balance leader one by one

**Comment by darionyaphet** [comment]
> Good job, several questions, would you post it in summary:
> 
> 1. What is the new balance command?
> 2. Does `balance leader` logic changed anything? Because I see you moved to another `LeaderBalanceProcessor.cpp`. If any changes, please tell me.

1. new balance command is `SUBMIT JOB DATA BALANCE` and `SUBMIT JOB DATA BALANCE REMOVE ${host}`
2. The logic doesn't change, it just changes places

File: src/meta/processors/admin/LeaderBalanceProcessor.cpp (hunk 2 above)
Comment (critical27) [reply src/meta/processors/admin/LeaderBalanceProcessor.cpp:28]: We'd better keep it. For example if `balance leader space` then we only balance for that space, if `balance leader` then all spaces.

**Comment by critical27** [comment]
Ready to test?

**Comment by darionyaphet** [comment]
> Ready to test?

Fire

**Pull request closed without merging**